File: Embarcadero/XE5/CPP/VideoPlayback/VideoForm.cpp
//---------------------------------------------------------------------------

#include <fmx.h>
#include <System.ioutils.hpp>
#pragma hdrstop

#include "VideoForm.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.fmx"
TVideoPlayBackForm *VideoPlayBackForm;
//---------------------------------------------------------------------------
__fastcall TVideoPlayBackForm::TVideoPlayBackForm(TComponent* Owner)
	: TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TVideoPlayBackForm::videoBtnClick(TObject *Sender)
{
  /*Under Project-Deployment, we added the media file and set the remote path.
  When the program starts, everything in the directory that is set under remote path is
  copied over to the Documents folder on the device. The MediaPlayer loads the file from
  the Documents folder*/
#if defined(TARGET_OS_IPHONE) || defined(TARGET_IPHONE_SIMULATOR)
	MediaPlayer1->FileName =
		IncludeTrailingPathDelimiter(System::Ioutils::TPath::GetDocumentsPath()) + "Ocean.mp4";
#endif
	MediaPlayer1->Play();
}
//---------------------------------------------------------------------------

File: Embarcadero/XE7/ObjectPascal/VideoPlayback/VideoForm.pas
unit VideoForm;

interface

uses
  System.SysUtils, System.Types, System.UITypes, System.Classes, System.Variants,
  FMX.Types, FMX.Controls, FMX.Forms, FMX.Dialogs, FMX.Media, FMX.StdCtrls;

type
  TVideoPlayBackForm = class(TForm)
    videoBtn: TButton;
    MediaPlayer1: TMediaPlayer;
    ToolBar1: TToolBar;
    Label1: TLabel;
    procedure videoBtnClick(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  VideoPlayBackForm: TVideoPlayBackForm;

implementation

{$R *.fmx}
{$R *.LgXhdpiPh.fmx ANDROID}

uses
  System.iOUtils;

type
  TOpenMedia = class(TMedia);

procedure TVideoPlayBackForm.videoBtnClick(Sender: TObject);
begin
  {Under Project-Deployment, we added the media file and set the remote path.
  When the program starts, everything in the directory that is set under remote path is
  copied over to the Documents folder on the device. The MediaPlayer loads the file from
  the Documents folder}
  MediaPlayer1.FileName := IncludeTrailingPathDelimiter(TPath.GetDocumentsPath) + 'Ocean.mp4';
  MediaPlayer1.Play;
end;

end.
